test/pstl_offload/memory/allocation_utils.h

```diff
@@ -0,0 +1,119 @@
+#endif // __linux__
+
+    void *new_ptr;
+    void *arr_new_ptr;
+    void *nothrow_new_ptr;
+    void *arr_nothrow_new_ptr;
+    void *aligned_new_ptr;
+    void *aligned_nothrow_new_ptr;
+    void *aligned_arr_new_ptr;
+    void *aligned_nothrow_arr_new_ptr;
+
+    static const std::size_t alloc_size = 1024;
+    static const std::size_t alignment = 16;
+};
+
+static void perform_allocations_impl(allocs& na) {
+    na.malloc_ptr = malloc(allocs::alloc_size);
+    na.calloc_ptr = calloc(allocs::alloc_size, allocs::alloc_size);
+    na.realloc_ptr = realloc(nullptr, allocs::alloc_size);
+#if __linux__
+    EXPECT_TRUE(malloc_usable_size(na.malloc_ptr) >= allocs::alloc_size, "Invalid object size");
+    na.memalign_ptr = memalign(allocs::alignment, allocs::alloc_size);
+    posix_memalign(&na.posix_memalign_ptr, allocs::alignment, allocs::alloc_size);
+    na.aligned_alloc_ptr = aligned_alloc(allocs::alignment, allocs::alloc_size);
+    na.libc_malloc_ptr = __libc_malloc(allocs::alloc_size);
+    na.libc_calloc_ptr = __libc_calloc(allocs::alloc_size, allocs::alloc_size);
+    na.libc_realloc_ptr = __libc_realloc(nullptr, allocs::alloc_size);
+    na.libc_memalign_ptr = __libc_memalign(16, allocs::alloc_size);
+#endif // __linux__
+
+    na.new_ptr = ::operator new(allocs::alloc_size);
+    na.arr_new_ptr = ::operator new[](allocs::alloc_size);
+    na.nothrow_new_ptr = ::operator new(allocs::alloc_size, std::nothrow);
+    na.arr_nothrow_new_ptr = ::operator new[](allocs::alloc_size, std::nothrow);
+    na.aligned_new_ptr = ::operator new(allocs::alloc_size, std::align_val_t(allocs::alignment));
+    na.aligned_nothrow_new_ptr = 
+        ::operator new(allocs::alloc_size, std::align_val_t(allocs::alignment), std::nothrow);
+    na.aligned_arr_new_ptr = ::operator new[](allocs::alloc_size, std::align_val_t(allocs::alignment));
+    na.aligned_nothrow_arr_new_ptr =
+        ::operator new[](allocs::alloc_size, std::align_val_t(allocs::alignment), std::nothrow);
+}
+
+static void perform_deallocations_impl(const allocs& na) {
+    free(na.malloc_ptr);
+    free(na.calloc_ptr);
+    free(na.realloc_ptr);
+#if __linux__
+    free(na.memalign_ptr);
+    free(na.posix_memalign_ptr);
+    free(na.aligned_alloc_ptr);
+    free(na.libc_malloc_ptr);
+    free(na.libc_calloc_ptr);
+    free(na.libc_realloc_ptr);
+    free(na.libc_memalign_ptr);
+#endif // __linux__
+
+    operator delete(na.new_ptr);
+    operator delete[](na.arr_new_ptr);
+    operator delete(na.nothrow_new_ptr, allocs::alloc_size);
+    operator delete[](na.arr_nothrow_new_ptr, allocs::alloc_size);
+    operator delete(na.aligned_new_ptr, std::align_val_t(allocs::alignment));
+    operator delete(na.aligned_nothrow_new_ptr, std::align_val_t(allocs::alignment));
+    operator delete[](na.aligned_arr_new_ptr, std::align_val_t(allocs::alignment));
+    operator delete[](na.aligned_nothrow_arr_new_ptr, std::align_val_t(allocs::alignment));
+}
+
+void perform_system_allocations(allocs&);
```

Comment:
How about applying the following suggestion to have a more symmetric API with malloc/free? 
```suggestion
allocs perform_system_allocations();
```

Reply:
Done.